api/deployments/models/component_builder.go

```diff
@@ -119,13 +119,9 @@ func (b *componentBuilder) WithComponent(component radixv1.RadixCommonDeployComp
 	b.secrets = component.GetSecrets()
 
 	for _, volumeMount := range component.GetVolumeMounts() {
-		volumeMountType := deployment.GetCsiAzureVolumeMountType(&volumeMount)
+		volumeMountType := volumemount.GetCsiAzureVolumeMountType(&volumeMount)
 		switch volumeMountType {
-		case radixv1.MountTypeBlob:
-			secretName := defaults.GetBlobFuseCredsSecretName(component.GetName(), volumeMount.Name)
-			b.secrets = append(b.secrets, secretName+defaults.BlobFuseCredsAccountKeyPartSuffix)
-			b.secrets = append(b.secrets, secretName+defaults.BlobFuseCredsAccountNamePartSuffix)
-		case radixv1.MountTypeBlobFuse2FuseCsiAzure, radixv1.MountTypeBlobFuse2Fuse2CsiAzure, radixv1.MountTypeBlobFuse2NfsCsiAzure, radixv1.MountTypeAzureFileCsiAzure:
+		case radixv1.MountTypeBlobFuse2FuseCsiAzure, radixv1.MountTypeBlobFuse2Fuse2CsiAzure:
```

Comment:
We should not app account name and key secrets when using workload identity.
Also, if the user has defined storage account name in radixconfig (without workload identity), then we should not add the secret defined by suffix CsiAzureCredsAccountNamePartSuffix

Reply:
Removed secrets in such cases, also preventing secrets to be potentially added to emptydir volume mount